pyanaconda/ui/gui/spokes/lib/dasdfmt.py

```diff
@@ -108,9 +86,24 @@ def on_format_clicked(self, *args):
         self._ok_button.set_sensitive(False)
         self._notebook.set_current_page(1)
 
+        # Format dasds and update the storage.
+        threadMgr.add(AnacondaThread(name=constants.THREAD_DASDFMT, target=self.run_format, args=()))
+
+    def run_format(self):
+        """Run the dasd formatting and update the storage."""
+        epoch_started = self._epoch
+
         # Loop through all of our unformatted DASDs and format them
-        threadMgr.add(AnacondaThread(name=constants.THREAD_DASDFMT,
-                                target=self.run_dasdfmt, args=(self._epoch,)))
+        self.dasds_formatting.report.connect(self.show_dasdfmt_report)
+        self.dasds_formatting.run(self.storage, self.data)
+        self.dasds_formatting.report.disconnect(self.show_dasdfmt_report)
```

Comment:
I really don't like this access to `dasd_formatting` public variable on different thread. This should be separated or locked somehow or something in future will change and we will start to get synchronization problems.

Reply:
Fixed.